localstack/utils/archives.py

```diff
@@ -185,12 +185,29 @@ def download_and_extract(archive_url, target_dir, retries=0, sleep=3, tmp_archiv
     if not os.path.exists(tmp_archive) or os.path.getsize(tmp_archive) <= 0:
         # create temporary placeholder file, to avoid duplicate parallel downloads
         save_file(tmp_archive, "")
-        for i in range(retries + 1):
+
+        current_try = 1
+        while current_try <= retries + 1:
```

Comment:
nit: I think it's a bit more concise to use the `for` loop (as it was before):
```
        for current_try in range(retries + 1):
```
They are easier to read, and you can spare a few lines (the assignment and increment of `current_try`).